packages/dialect-react/components/DialectContext/index.tsx

```diff
@@ -210,6 +211,18 @@ export const DialectProvider = (props: PropsType): JSX.Element => {
     fetchError?.type,
   ]);
 
+  useEffect(() => {
+    if (messages.length > 0) {
```

Comment:
There are some repeating conditions here, I suggest we simplify this a bit:

Some deps were missing as well, added them to the array too. Please do check that it's working the same way as your current implementation (this was a blind refactor... might've messed it up).

```
useEffect(() => {
  if (!wallet || !dialect?.dialect) {
    setMessages([]);
    return;
  }

  if (messages.length !== dialect.dialect.messages.length || messages.length === 0) {
    setMessages(dialect.dialect.messages);
  }
}, [wallet, dialect?.dialect, messages.length]);
```

Let me know, what you think. Thanks!

Reply:
Actually, I think it can be as simple as this:

```
useEffect(() => {
    const hasNewMessage = wallet && dialect?.dialect && messages.length !== dialect.dialect.messages.length;
    if (hasNewMessage) {
      setMessages(dialect.dialect.messages);
    }
  }, [
    wallet,
    dialect?.dialect,
    messages.length,
  ]);
```
We just want to update the messages if `dialect.dialect.messages.length` differs from our current `messages.length` but we don't want to update it if `dialect` is `undefined`

Reply:
Even better, great job!